### PR TITLE
Add styling to the WFO page

### LINKED ISSUE #1474: WFO Page template
Using the following [Figma file](https://www.figma.com/design/w2QI2uTgaTWUPLHLi40mm1/Weather.gov-%E2%80%93-v0.2?node-id=9942-4943&t=vX6pzHVfmYJxkin4-0) implement the front-end styling for the WFO page.

### PR DESCRIPTION
**Pull request by greg-does-weather**
## What does this PR do? 🛠️

This extends the previous work standing up a WFO page by adding the styling described in [Figma](https://www.figma.com/design/w2QI2uTgaTWUPLHLi40mm1/Weather.gov-%E2%80%93-v0.2?node-id=9942-4943&t=X9aYfYrgCCeLJTLF-0).

This PR makes one extension to the previous approach to rendering WFO pages. If an office has not create a WFO information content node, we still show that WFO's page using the information available to us – its name and coverage area map. Requests for WFO pages for offices that don't exist continues to result in a 404.

closes #1474

## Screenshots (if appropriate): 📸

<details>
<summary>Full content, mobile size</summary>

</details>

<details>
<summary>Full content, desktop size</summary>

</details>

<details>
<summary>Default content, mobile size</summary>

</details>

<details>
<summary>Default content, desktop size</summary>

</details>

### CODE REVIEW

**Review by eric-gade — APPROVED** [review]
This looks good and works well on my end 👍 
  
Just a couple of things:
  
Renaming the individual field templates means those fields will display the same way where ever they are (re)used. I think this is fine, but something to consider when building out the WFO Promo block, for example, where we can probably use some kind of drupal shortcuts to just render the fields according to existing templates if we want.
  
I'll leave most of the styling stuff to @partly-igor , but I did notice that there probably needs to be some margin on the bottom in the mobile/tablet view near the forecast discussion (though I know that is work that is forthcoming):
  
Screenshot 2024-08-12 at 9 59 45 AM

## Code Review Checklist

This is an automated comment on every pull request requiring a review. A checked item is either an assertion that I have tested this item or an indication that I have verified it does not apply to this pull request.

## The Basics
- [x] Checks are passing
- [x] I read the code
- [x] I ran the code
- [x] (if applicable) Post deploy steps are run
- [x] (if applicable) I validated the change on the deployed version in <environment>

## Documentation
- [x] changes to “how we do things” are documented in READMEs
- [x] all new functions and methods are commented using plain language
- [x] any new modules added documented in modules.md

## Security
- [x] security false positives are documented
- [x] data from external sources is cleaned and clearly marked

## Reliability
- [x] error handling exists for unusual or missing values
- [x] interactions with external systems are wrapped in try/except
- [x] functionality is tested with unit or integration tests
- [x] dependency updates in composer.json also got changed in composer-lock.json

## Infrastructure
- [x] all changes are auditable and documented via a script
- [x] it is clear who can and should run the script
- [x] (if applicable) diagrams have been updated or added in PlantUML

## Accessibility
- [x] New pages have been added to cypress-axe file so that they will be tested with our automated accessibility testing
- [x] Meets WCAG 2.0 AA or 2.1 AA for Section 508 compliance
    - [ ] Site is keyboard accessible. All interactions can be accessed with a keyboard
    - [ ] Site is free of keyboard traps. The keyboard focus is never trapped in a loop
    - [ ] All form inputs have explicit labels
    - [ ] Form instructions are associated with inputs
    - [ ] All relevant images use an img tag
    - [ ] All images have appropriate alt attributes
    - [ ] Multimedia is tagged. All multimedia has appropriate captioning and audio description
    - [ ] Text has sufficient color contrast. All text has a contrast ratio of 4.5:1 with the background
    - [ ] Site never loses focus. Focus is always visible when moving through the page with the keyboard
    - [ ] Tab order is logical
    - [ ] Tables are coded properly. Tables have proper headers and column attributes
    - [ ] Headings are nested properly. Heading elements are nested in a logical way
    - [ ] Language is set. The language for the page is set
    - [ ] CSS is not required to use the page. The page makes sense with or without CSS
    - [ ] Links are unique and contextual. All links can be understood taken alone, e.g., ‘Read more - about 508’
    - [ ] Page titles are descriptive

## Device Matrix
- [x] firefox/gecko (renders correctly and user interactions work)
- [x] chrome/chromium/edge (renders correctly and user interactions work)
- [ ] safari/webkit (renders correctly and user interactions work)
- [x] web page is readable and usable 
    - [ ] at 480px     (mobile)
    - [ ] at 640px     (tablet)
    - [ ] at 1024px    (desktop)

File: spatial-data/maps/template.html [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            opacity: 0.85,
+            fillOpacity: 0.3,
+          },
+        }).addTo(map);
+
+        // Compute the bounds, center the map accordingly, and then zoom out
+        // one step so we have a somewhat broader view.
+        const bounds = wfo.getBounds();
+        map.fitBounds(bounds);
+        map.zoomOut(1);
```
Comment (partly-igor) [inline spatial-data/maps/template.html:60]: I am curious to see what it looks like without this one step zoom out – not sure what the right level is, but I am wondering if the current one is a touch too zoomed out.

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig [hunk 2]
```diff
@@ -1,41 +1,111 @@
...
+      {{ content.field_wfo_image }}
+    </div>
+  {% endif %}
+
+  <div class="grid-container">
+    <h1 class="text-normal text-primary-darker">
+      {{ "@wfoName (@wfoCode) National Weather Service Office" | t({ "@wfoName": wfo.name, "@wfoCode": wfo.code }) }}
+    </h1>
+
+    <div class="grid-row">
```
Comment (partly-igor) [inline web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig:13]: add `grid-gap` class to create the right gap between the two columns

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig [hunk 3]
```diff
@@ -1,41 +1,111 @@
...
+      {% set hasAbout = not (content.body.isEmpty or content.body.0 is not defined) %}
+      {% if hasAbout %}
+      <div class="grid-col-12 tablet:grid-col-4">
+        <h2 class="text-normal text-primary-darker">{{ "About" | t }}</h2>
+        {{content.body}}
+      </div>
+      {% endif %}
+
+      <div class="grid-col-12{% if hasAbout %} tablet:grid-col-8 tablet:padding-left-2{% endif %}">
```
Comment (partly-igor) [inline web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig:22]: with `grid-gap` added I think you could remove `tablet:padding-left-2`

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig [hunk 4]
```diff
@@ -1,41 +1,111 @@
...
+  <div class="grid-container">
+    <h1 class="text-normal text-primary-darker">
+      {{ "@wfoName (@wfoCode) National Weather Service Office" | t({ "@wfoName": wfo.name, "@wfoCode": wfo.code }) }}
+    </h1>
+
+    <div class="grid-row">
+      {% set hasAbout = not (content.body.isEmpty or content.body.0 is not defined) %}
+      {% if hasAbout %}
+      <div class="grid-col-12 tablet:grid-col-4">
+        <h2 class="text-normal text-primary-darker">{{ "About" | t }}</h2>
```
Comment (partly-igor) [inline web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig:17]: Want to align "About" to top of map, so add `margin-top-0`

Comment (partly-igor) [inline web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig:17]: For some reason all H2s on this page are rendering at 24px, they should be `xl` or ~28px

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig [hunk 5]
```diff
@@ -1,41 +1,111 @@
...
+    #}
+    <div class="grid-row">
+      <div class="grid-col-12 tablet:grid-col-6">
+        <h2 class="border-top-1px border-base-light text-normal text-primary-darker padding-top-1">
+          {{ "Local expertise" | t }}
+        </h2>
+        <p>
+          Decision support is description tk tk tk
+        </p>
+        <ul class="padding-left-1">
```
Comment (partly-igor) [inline web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig:42]: Changing this to `padding-left-2` will optically align the bullets with the text

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig [hunk 6]
```diff
@@ -1,41 +1,111 @@
...
+          <li class="padding-y-05"><a href="#">Briefing</a></li>
+          <li class="padding-y-05"><a href="#">County finder</a></li>
+          <li class="padding-y-05"><a href="#">State page</a></li>
+          <li class="padding-y-05"><a href="#">Resource 1</a></li>
+          <li class="padding-y-05"><a href="#">Resource 2</a></li>
+          <li class="padding-y-05"><a href="#">Resource 3</a></li>
```
Comment (partly-igor) [inline web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig:48]: All links should have a class of `usa-link`

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig [hunk 7]
```diff
@@ -1,41 +1,111 @@
...
+
+      <div class="grid-col-12 tablet:grid-col-6">
+        <h2 class="border-top-1px border-base-light text-normal text-primary-darker padding-top-1">
+          {{ "Forecaster's discussion" | t }}
+        </h2>
+        <p>
+          The purpose of the area forecast discussion is to provide the
+          meteorological/scientific reasoning behind the forecast
+        </p>
+        <a href="#">Current Area Forecast Discussion</a>
```
Comment (partly-igor) [inline web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig:60]: link should have a class of `usa-link`

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig [hunk 8]
```diff
@@ -1,41 +1,111 @@
...
+
+          {% if hasEmail %}
+            {% if hasPhone %}<div class="margin-top-2">{% endif %}
+              {{ content.field_wfo_email }}
+            {% if hasPhone %}</div>{% endif %}
+          {% endif %}
+
+          {% if hasAddress %}
+            {% if hasPhone or hasEmail %}<div class="margin-top-2">{% endif %}
+              {{ content.field_address }}
```
Comment (partly-igor) [inline web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig:94]: address field should have `margin-top-0`

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig [hunk 9]
```diff
@@ -1,41 +1,111 @@
...
+        </figure>
+      </div>
+    </div>
+
+    {#
+      TODO: This is all placeholder stuff. When it comes out in favor of real
+      stuff, make sure we add things to the translations list appropriately.
+    #}
+    <div class="grid-row">
+      <div class="grid-col-12 tablet:grid-col-6">
```
Comment (partly-igor) [inline web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig:35]: On mobile the spacing between the sections is a uneven. Could we make it so there is 4 units of space between the between the bottom of one section and the start (border) of the next

Screenshot of uneven spacing

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig [hunk 10]
```diff
@@ -1,41 +1,111 @@
...
+      <div class="grid-col-12 tablet:grid-col-4">
+        <h2 class="text-normal text-primary-darker">{{ "About" | t }}</h2>
+        {{content.body}}
+      </div>
+      {% endif %}
+
+      <div class="grid-col-12{% if hasAbout %} tablet:grid-col-8 tablet:padding-left-2{% endif %}">
+        <figure class="margin-0">
+          <img src="/{{ directory }}/assets/images/wfos/{{ wfo.code }}.png" alt="">
+          <figcaption>{{ "Coverage area for @wfo WFO" | t({ "@wfo": wfo.name })}}</figcaption>
```
Comment (partly-igor) [inline web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig:25]: The figure text should be a bit smaller `class="font-sans-xs"`

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig [hunk 11]
```diff
@@ -1,41 +1,111 @@
...
-</dl>
+<div class="wx-wfo-info">
+  {% if content.field_wfo_image is not empty %}
+    <div class="wx-wfo-image grid-container padding-0 tablet:padding-x-2 desktop:padding-x-4">
+      {{ content.field_wfo_image }}
+    </div>
+  {% endif %}
+
+  <div class="grid-container">
+    <h1 class="text-normal text-primary-darker">
```
Comment (partly-igor) [inline web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig:9]: Something I missed, but I think we need a little extra space below the h1 on desktop. Could you add `desktop:margin-bottom-3`

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig [hunk 12]
```diff
@@ -1,41 +1,111 @@
...
+              {{ content.field_facebook_url }}
+              {{ content.field_twitter_url }}
+              {{ content.field_youtube_url }}
+            </ul>
+          {% endif %}
+        </div>
+      </div>
+    {% endif %}
+  </div>
+</div>
```
Comment (partly-igor) [inline web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig:111]: As Eric notes there should be 10 units of space at the bottom of the page before the footer

**Review by partly-igor — CHANGES_REQUESTED** [review]
Looking good, there are some design/style updates to make it match the design.

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig (hunk 2 above)
Comment (greg-does-weather) [reply web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig]: 👍

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig (hunk 3 above)
Comment (greg-does-weather) [reply web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig]: 👍

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig (hunk 4 above)
Comment (greg-does-weather) [reply web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig]: 👍

Comment (greg-does-weather) [reply web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig]: 👍

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig (hunk 5 above)
Comment (greg-does-weather) [reply web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig]: 👍

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig (hunk 6 above)
Comment (greg-does-weather) [reply web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig]: 👍

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig (hunk 8 above)
Comment (greg-does-weather) [reply web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig:94]: 👍

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig (hunk 7 above)
Comment (greg-does-weather) [reply web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig]: 👍

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig (hunk 9 above)
Comment (greg-does-weather) [reply web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig:35]: 👍 I added `margin-top-4 padding-top-4` to all the headings and that seemed to accomplish it?

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig (hunk 10 above)
Comment (greg-does-weather) [reply web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig]: 👍

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig (hunk 11 above)
Comment (greg-does-weather) [reply web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig]: 👍

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig (hunk 12 above)
Comment (greg-does-weather) [reply web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig:111]: 👍

**Comment by partly-igor** [comment]
Nice yeah, I think that without the zoom out is a good balance.

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig (hunk 9 above)
Comment (partly-igor) [reply web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig:35]: Yep, that looks great, only one that sticks out is the list under Local Expertise, so I think we add `margin-bottom-0` to that `ul` we should be good to go

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig [hunk 13]
```diff
@@ -33,31 +33,31 @@
     #}
     <div class="grid-row">
       <div class="grid-col-12 tablet:grid-col-6">
-        <h2 class="border-top-1px border-base-light text-normal text-primary-darker padding-top-1">
+        <h2 class="font-sans-xl border-top-1px border-base-light text-normal text-primary-darker padding-top-4 margin-top-4">
```
Comment (partly-igor) [inline web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig:36]: `padding-top-105`

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig [hunk 14]
```diff
@@ -33,31 +33,31 @@
...
+          <li class="padding-y-05"><a class="usa-link" href="#">State page</a></li>
+          <li class="padding-y-05"><a class="usa-link" href="#">Resource 1</a></li>
+          <li class="padding-y-05"><a class="usa-link" href="#">Resource 2</a></li>
+          <li class="padding-y-05"><a class="usa-link" href="#">Resource 3</a></li>
         </ul>
       </div>
 
       <div class="grid-col-12 tablet:grid-col-6">
-        <h2 class="border-top-1px border-base-light text-normal text-primary-darker padding-top-1">
+        <h2 class="font-sans-xl border-top-1px border-base-light text-normal text-primary-darker padding-top-4 margin-top-4">
```
Comment (partly-igor) [inline web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig:53]: `padding-top-105`

File: web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig [hunk 15]
```diff
@@ -73,7 +73,7 @@
     {% set hasContact = hasSocial or hasAddress or hasPhone or hasEmail %}
 
     {% if hasContact %}
-      <h2 class="border-top-1px border-base-light text-normal text-primary-darker padding-top-1 margin-bottom-0">
+      <h2 class="font-sans-xl border-top-1px border-base-light text-normal text-primary-darker margin-bottom-0 padding-top-4 margin-top-4">
```
Comment (partly-igor) [inline web/themes/new_weather_theme/templates/wfo-info/node--wfo-info.html.twig:76]: `padding-top-105`

**Review by partly-igor — COMMENTED** [review]
Updates look really good, except the padding below the border should be `105` instead of `4`

**Review by partly-igor — APPROVED** [review]
Updates look good!

**Pull request merged**